### PR TITLE
Better reports from logger

### PR DESCRIPTION
**Pull request by mitchellhenke**
This should address #101, #117, and make it somewhat easier to make improvements going forward.  It should also help disambiguate errors from Processes.

@jeregrine

### CODE REVIEW

File: lib/sentry/event.ex [hunk 1]
```diff
@@ -107,15 +107,30 @@ defmodule Sentry.Event do
   """
   @spec transform_exception(Exception.t, keyword()) :: Event.t
   def transform_exception(exception, opts) do
+    error_type = Keyword.get(opts, :error_type) || :error
     normalized = Exception.normalize(:error, exception)
 
+    type = case error_type do
```
Comment (jeregrine) [inline lib/sentry/event.ex]: could also do an if/else here

File: lib/sentry/event.ex [hunk 2]
```diff
@@ -107,15 +107,30 @@ defmodule Sentry.Event do
...
+      :error ->
+        normalized.__struct__
+      _ ->
+        error_type
+    end
+
+    value = case error_type do
```
Comment (jeregrine) [inline lib/sentry/event.ex]: same

File: lib/sentry/event.ex [hunk 3]
```diff
@@ -107,15 +107,30 @@ defmodule Sentry.Event do
...
+      _ ->
+        Exception.format_banner(error_type, exception)
+    end
+
+    module = Keyword.get(opts, :module)
+    exception = [%{type: type, value: value, module: module}]
     message = :error
               |> Exception.format_banner(normalized)
               |> String.trim("*")
               |> String.trim
```
Comment (jeregrine) [inline lib/sentry/event.ex]: parens()

File: lib/sentry/logger.ex [hunk 4]
```diff
@@ -21,40 +21,45 @@ defmodule Sentry.Logger do
...
+
+  defp get_exception_and_stacktrace({kind, {exception, sub_stack}, _stack}) when is_list(sub_stack) do
+    {kind, exception, sub_stack}
+  end
+  defp get_exception_and_stacktrace({kind, exception, stacktrace}) do
+    {kind, exception, stacktrace}
+  end
+
+  defp get_initial_call_and_module({kind, exception, stacktrace}, error_info) do
+    case Keyword.get(error_info, :initial_call) do
```
Comment (jeregrine) [inline lib/sentry/logger.ex]: whats going on here?

Comment (mitchellhenke) [inline lib/sentry/logger.ex]: I need to add some documentation around this.  Sentry primarily groups by stacktrace, and error reports will include an `:initial_call` that's generally the MFA of how the Process was started, e.g. `{MyModule.GenServer, :init, [:arg]`.  For bad exits or throws, the stacktrace is the exact same regardless of which module it came from, so they ended up being grouped together.  Something like:

```elixir
[{:gen_server, :terminate, 7, [file: 'gen_server.erl', line: 812]},
 {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 247]}]
```

Comment (jeregrine) [inline lib/sentry/logger.ex]: Ahh I see

**Comment by mitchellhenke** [comment]
@jeregrine updated

**Comment by jeregrine** [comment]
cc @fishcakez

**Pull request merged**